src/crafting.cpp

```diff
@@ -1457,76 +1127,50 @@ bool player::disassemble( int dis_pos )
-        dis_item.calc_rot( global_square_location() );
-        if( dis_item.rotten() ||
-            ( dis_item.is_food_container() && dis_item.contents.front().rotten() ) ) {
-            if( msg_and_query ) {
-                add_msg( m_info, _( "It's rotten, I'm not taking that apart." ) );
+    // refuse to disassemble rotten items
+    if( obj.goes_bad() || ( obj.is_food_container() && obj.contents.front().goes_bad() ) ) {
+        obj.calc_rot( global_square_location() );
+        if( obj.rotten() || ( obj.is_food_container() && obj.contents.front().rotten() ) ) {
+            if( interactive ) {
+                add_msg_if_player( m_info, _( "It's rotten, I'm not taking that apart." ) );
             }
             return false;
         }
     }
 
-    // First check regular disassembly, then book "fake disassembly"
-    // Note: this may get weird if books ever get regular disassembly
-    std::string recipe_ident;
-    int recipe_time = 100;
-    const inventory &crafting_inv = crafting_inventory();
-    if( cur_recipe != nullptr &&
-        can_disassemble( dis_item, cur_recipe, crafting_inv, msg_and_query ) ) {
-        if( msg_and_query && !query_disassemble( dis_item ) ) {
-            return false;
-        }
-
-        recipe_ident = cur_recipe->ident();
-        recipe_time = cur_recipe->time;
-    }
-    // If we're trying to disassemble a book or magazine
-    if( dis_item.is_book() ) {
-        if( msg_and_query && get_option<bool>( "QUERY_DISASSEMBLE" ) &&
-            !query_yn( _( "Do you want to tear %s into pages?" ),
-                       dis_item.tname().c_str() ) ) {
-            return false;
-        } else {
-            recipe_ident = fake_recipe_book;
-        }
+    // check sufficient tools for disassembly
+    const inventory &inv = crafting_inventory();
+    if( !can_disassemble( obj, inv, interactive ) ) {
+        return false;
     }
 
-    if( recipe_ident.empty() ) {
-        // No recipe exists, or the item cannot be disassembled
-        if( msg_and_query ) {
-            add_msg( m_info, _( "The %s cannot be disassembled!" ),
-                     dis_item.tname().c_str() );
-        }
+    // last chance to back out
+    if( interactive && get_option<bool>( "QUERY_DISASSEMBLE" ) &&
+        !query_yn( _( "Really disassemble the %s?" ), obj.tname().c_str() ) ) {
         return false;
     }
 
     if( activity.type != ACT_DISASSEMBLE ) {
-        assign_activity( ACT_DISASSEMBLE, recipe_time );
+        assign_activity( ACT_DISASSEMBLE, r.time );
     } else if( activity.moves_left <= 0 ) {
-        activity.moves_left = recipe_time;
+        activity.moves_left = r.time;
     }
 
-    activity.values.push_back( dis_pos );
-    activity.coords.push_back( ground ? pos() : tripoint_min );
-    activity.str_values.push_back( recipe_ident );
+    activity.values.push_back( pos );
+    activity.coords.push_back( ground ? this->pos() : tripoint_min );
```

Comment:
Because `pos` is unfortunately already a parameter. I want to rip that section out in a future PR so leaving as-is for now.